pkg/ring/tokens.go

```diff
@@ -0,0 +1,99 @@
+package ring
+
+import (
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"os"
+
+	"github.com/cortexproject/cortex/pkg/util"
+	"github.com/go-kit/kit/log/level"
+	"github.com/prometheus/prometheus/tsdb/fileutil"
+)
+
+const (
+	// TokensVersion1 is the version is a simple list of tokens.
+	TokensVersion1 = 1
+)
+
+// Tokens is a simple list of tokens.
+type Tokens []uint32
+
+// StoreToFile stores the tokens in the given directory.
+func (t Tokens) StoreToFile(tokenFilePath string) error {
+	if tokenFilePath == "" {
+		return errors.New("path is empty")
+	}
+
+	f, err := ioutil.TempFile(os.TempDir(), "tokens")
+	if err != nil {
+		return err
+	}
+
+	defer func() {
+		// If the file was not closed, then there must already be an error, hence ignore
+		// the error (if any) from f.Close(). If the file was already closed, then
+		// we would ignore the error in that case too.
+		f.Close()
+		// RemoveAll returns no error when tmp doesn't exist so it is safe to always run it.
+		if err := os.RemoveAll(f.Name()); err != nil {
```

Comment:
Do we need to delete the temp file? If writing of the token fails, I think it would be better to keep the temp file around, so that it can be inspected by operator and moved to target 'tokens' file. So basically, either temp file is renamed to target location, or we keep it. WDYT?

Reply:
Makes sense, we can keep the temporary file hanging around.